lib/chef/provider/service/solaris.rb

```diff
@@ -56,7 +56,8 @@ def define_resource_requirements
 
         def enable_service
           shell_out!(default_init_command, "clear", @new_resource.service_name) if @maintenance
-          shell_out!(default_init_command, "enable", "-s", @new_resource.service_name)
+          start_flags = @new_resource.recursive ? "-sr" : "-s"
+          shell_out!(default_init_command, "enable", start_flags , @new_resource.service_name)
```

Comment:
can you just pass `@new_resource.options` here directly instead so that arbitrary options can be passed -- same presumably with other actions.